packages/web3-eth-contract/types/index.d.ts

```diff
@@ -0,0 +1,191 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import BN = require('bn.js');
+import {
+    PromiEvent,
+    provider,
+    TransactionSigner,
+    Web3ModuleOptions
+} from 'web3-core';
+import { AbiInput, AbiItem, AbiOutput } from 'web3-utils';
+
+export class Contract {
+    constructor(
+        provider: provider,
+        abi: AbiItem[],
+        address?: string,
+        options?: ContractOptions
+    );
+
+    address: string;
+    jsonInterface: AbiModel;
```

Comment:
The class AbiModel doesn't exist in 1.0

Reply:
AbiModel is not a class its a interface and still looks correct when comparing object to what 1.0 json interface object is.